src/puppetlabs/puppetdb/query_eng.clj

```diff
@@ -86,8 +63,35 @@
   If the query can't be parsed, a 400 is returned."
   [entity version query paging-options db url-prefix]
   (try
-    (let [parsed-query (json/parse-strict-string query true)]
-      (stream-query-result entity version parsed-query paging-options db url-prefix http/stream-json-response))
+    (let [parsed-query (json/parse-strict-string query true)
+          query-error (promise)
+          count-query (promise)
```

Comment:
"Include total" is used for pagination, its a guide marker to see how many total records there are, so UI devs can produce a maximum page link in their UI.